tests/data_context/test_data_context_datasources.py

```diff
@@ -427,3 +428,32 @@ def test_BaseDataContext_delete_datasource_updates_cache(
 
     assert not mock_delete.called
     assert name not in context.datasources
+
+@pytest.mark.unit
+def test_list_datasources() -> None:
+    project_config = DataContextConfig(
+        store_backend_defaults=InMemoryStoreBackendDefaults()
+    )
+    project_config.datasources = {
+        "my_datasource_name": {
+            "class_name": "Datasource",
+            "data_connectors": {},
+            "execution_engine": {
+                "class_name": "PandasExecutionEngine",
+                "module_name": "great_expectations.execution_engine"
+            },
+            "module_name": "great_expectations.datasource"
+        }
+    }
+    project_config.fluent_datasources = {
+        "fluent_datasources": {
+            "my_datasource_1": {
+                "type": "sqlite",
+                "name": "my_datasource_1",
+                "connection_string": "sqlite: // /"
+            }
+        }
+    }
+    context = gx.get_context(project_config=project_config)
+
+    assert len(context.list_datasources()) == 2
```

Comment:
If my earlier comment about credential obfuscation is relevant, could we add an additional test here that asserts against the actual output (checking to make sure that sensitive values are covered)

Reply:
yup